logstash-core/lib/logstash/agent_plugin_manager.rb

```diff
@@ -0,0 +1,65 @@
+# encoding: utf-8
+require "logstash/agent"
+
+module LogStash module AgentPluginManager
```

Comment:
Not the best one myself about naming, but I think I prefer if we name this class thing like ```AgentPluginRegistry``` what do you think?